File: backend/search-api/src/main/scala/no/ndla/searchapi/model/api/SearchResultDTO.scala
/*
 * Part of NDLA search-api
 * Copyright (C) 2018 NDLA
 *
 * See LICENSE
 *
 */

package no.ndla.searchapi.model.api

import sttp.tapir.Schema.annotations.description

@description("Information about search-results")
case class SearchResultDTO[T](
    @description("The total number of articles matching this query")
    totalCount: Long,
    @description("For which page results are shown from")
    page: Int,
    @description("The number of results per page")
    pageSize: Int,
    @description("The chosen search language")
    language: String,
    @description("The search results")
    results: Seq[T],
)

File: backend/taxonomy-api/src/main/kotlin/no/ndla/taxonomy/service/dtos/SearchResultDTO.kt
/*
 * Part of NDLA taxonomy-api
 * Copyright (C) 2022 NDLA
 *
 * See LICENSE
 */

package no.ndla.taxonomy.service.dtos

import io.swagger.v3.oas.annotations.media.Schema

@Schema(name = "SearchResult")
data class SearchResultDTO<T>(
    @field:Schema(description = "Total search result count, useful for fetching multiple pages")
    val totalCount: Long,
    @field:Schema(description = "The page number") val page: Int,
    @field:Schema(description = "The page size") val pageSize: Int,
    @field:Schema(description = "List of search results") val results: List<T>,
) {}
